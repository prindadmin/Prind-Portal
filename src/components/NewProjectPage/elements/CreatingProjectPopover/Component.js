import React, { Component } from 'react'
//import PropTypes from 'prop-types'

import PopOverHandler from '../../../common/popOverHandler'

import {
  Spinner,
  Intent,
} from '@blueprintjs/core'

import * as strings from '../../../../data/Strings'

export class Element extends Component {
  static propTypes = {
  }

  getContent = () => {
    return (
      <React.Fragment>
        {strings.PROJECT_CREATION_BILLING_WARNING}
        <Spinner size='150' intent={Intent.DANGER}/>
      </React.Fragment>
    )
  }

  render () {
    return (
      <PopOverHandler>
        <div id='popup-greyer' onClick={(e) => e.stopPropagation()}>
          <div id='project-creation-popover'>
            <div id='popup-box'>
              <div className='project-creation-popover-container' onClick={(e) => e.stopPropagation()}>
                <div className='element-title'>
                  {strings.PROJECT_CREATION_IN_PROGRESS}
                </div>
                <div className='element-description'>
                  {this.getContent()}
                </div>
              </div>
            </div>
          </div>
        </div>
      </PopOverHandler>
    )
  }

}

export default Element
