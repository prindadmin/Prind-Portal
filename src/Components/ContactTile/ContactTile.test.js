
import { shallow } from 'enzyme';
import Component from './ContactTile'
import ReactGA from 'react-ga';

var props;

beforeAll(() => {
  process.env = Object.assign(process.env, {
    REACT_APP_AWS_S3_USER_AVATAR_ENDPOINT: "https://prind-portal-user-profiles-dev.s3-eu-west-1.amazonaws.com/profile-avatar",
    REACT_APP_FUNCTIONALITY_USER_ACCREDITATIONS_LIST_V1: "True"
  });
})

beforeEach(() => {
  ReactGA.initialize('dummy', { testMode: true });
  props = {
    user: {
      username: "aaaa-aaaaaaaa-aaaa-aaaa-aaaa"
    },
    projects: {
      accessibleProjects: {
        projectCreator: [],
        projectRole: []
      },
      chosenProject: {
        projectName: "Test Project 1",
        projectId: "TestProjectID",
        projectType: "CDM2015",
      },
      memberList: {
        confirmed: [
          {
            username: "aaaa-aaaaaaaa-aaaa-aaaa-aaaa",
            roleID: 'creator',
          }
        ]
      },
      downloadURL: "",
      fileDetails: {},
      fetching: false,
      error: null,
    },
    memberDetails: {
      username: "bbbb-bbbbbbbb-bbbb-bbbb-bbbb",
      emailAddress: "testy.mctesterson@prind.tech",
      firstName: "Testy",
      lastName: "McTesterson",
    },
    onMemberRemove: jest.fn(),
    removeMember: jest.fn(),
    confirmed: false
  };
});

it('ContactTile - Should render', () => {
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});

it('ContactTile - Should render with confirmed user', () => {
  props.confirmed = true
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});

it('ContactTile - Should render without username', () => {
  props.memberDetails.username = undefined
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});

it('ContactTile - Should render without first and last name', () => {
  props.memberDetails.firstName = undefined
  props.memberDetails.lastName = undefined
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});

/*
it('ContactTile - Should render then unmount', () => {
  const component = shallow(<Component {...props} />);
  component.instance().componentWillUnmount()
  expect(component.state('loadingImage')).toEqual(expect.any(Function))
});
*/

it('ContactTile - Should remove member successfully', () => {
  const mockRemoveMember = jest.fn((projectId, username, resolve, reject) => {
    resolve()
  })
  const mockOnRemoveMember = jest.fn()
  props.removeMember = mockRemoveMember
  props.onMemberRemove = mockOnRemoveMember

  const component = shallow(<Component {...props} />);
  const removeButton = component.find('#remove-member-button')
  removeButton.simulate('click')
  expect(mockRemoveMember).toHaveBeenCalledTimes(1)
  expect(mockRemoveMember).toHaveBeenCalledWith("TestProjectID", "bbbb-bbbbbbbb-bbbb-bbbb-bbbb", expect.any(Function), expect.any(Function))
  expect(mockOnRemoveMember).toHaveBeenCalledTimes(1)
});

it('ContactTile - Should remove member failed', () => {
  const mockRemoveMember = jest.fn((projectId, username, resolve, reject) => {
    reject()
  })
  const mockOnRemoveMember = jest.fn()
  props.removeMember = mockRemoveMember
  props.onMemberRemove = mockOnRemoveMember

  const component = shallow(<Component {...props} />);
  const removeButton = component.find('#remove-member-button')
  removeButton.simulate('click')
  expect(mockRemoveMember).toHaveBeenCalledTimes(1)
  expect(mockRemoveMember).toHaveBeenCalledWith("TestProjectID", "bbbb-bbbbbbbb-bbbb-bbbb-bbbb", expect.any(Function), expect.any(Function))
  expect(mockOnRemoveMember).toHaveBeenCalledTimes(0)
});

it('ContactTile - Should click contact tile and show details', () => {
  const component = shallow(<Component {...props} />);
  const userTile = component.find('#contact-tile')
  userTile.simulate('click', { stopPropagation: jest.fn() })
  expect(component.state('showUserDetailsPopover')).toEqual(true)
});
