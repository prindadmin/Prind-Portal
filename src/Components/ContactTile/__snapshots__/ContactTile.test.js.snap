// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactTile - Should render 1`] = `
<div
  className="row member-not-confirmed"
  id="contact-tile"
  onClick={[Function]}
>
  <div
    className="col-md-3 col-sm-12"
  >
    <div
      className="text-center avatar-container"
    >
      <img
        alt="avatar"
        className="avatar img-circle img-thumbnail"
        src="images/default-avatar.png"
      />
    </div>
  </div>
  <div
    className="col-md-9 col-sm-12"
  >
    <div
      className="row"
    >
      <h4
        className="bp3-heading"
      >
        Testy McTesterson
      </h4>
    </div>
    <div
      className="row"
    >
      <div
        className="col-lg-2 col-md-4"
      >
        <b>
          Email: 
        </b>
      </div>
      <div
        className="col-lg-10 col-md-8"
      >
        testy.mctesterson@prind.tech
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-lg-2 col-md-4"
      >
        <b>
          Role: 
        </b>
      </div>
      <div
        className="col-lg-10 col-md-8"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="col-lg-2 col-md-4"
      >
        <b>
          Status: 
        </b>
      </div>
      <div
        className="col-lg-10 col-md-8"
      >
        Invited
      </div>
    </div>
    <div
      className="row"
    >
      <Blueprint3.Button
        id="remove-member-button"
        intent="danger"
        onClick={[Function]}
        text="Remove From Project"
      />
    </div>
  </div>
</div>
`;

exports[`ContactTile - Should render with confirmed user 1`] = `
<div
  className="row member-confirmed"
  id="contact-tile"
  onClick={[Function]}
>
  <div
    className="col-md-3 col-sm-12"
  >
    <div
      className="text-center avatar-container"
    >
      <img
        alt="avatar"
        className="avatar img-circle img-thumbnail"
        src="images/default-avatar.png"
      />
    </div>
  </div>
  <div
    className="col-md-9 col-sm-12"
  >
    <div
      className="row"
    >
      <h4
        className="bp3-heading"
      >
        Testy McTesterson
      </h4>
    </div>
    <div
      className="row"
    >
      <div
        className="col-lg-2 col-md-4"
      >
        <b>
          Email: 
        </b>
      </div>
      <div
        className="col-lg-10 col-md-8"
      >
        testy.mctesterson@prind.tech
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-lg-2 col-md-4"
      >
        <b>
          Role: 
        </b>
      </div>
      <div
        className="col-lg-10 col-md-8"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="col-lg-2 col-md-4"
      >
        <b>
          Status: 
        </b>
      </div>
      <div
        className="col-lg-10 col-md-8"
      >
        Joined
      </div>
    </div>
    <div
      className="row"
    >
      <Blueprint3.Button
        id="remove-member-button"
        intent="danger"
        onClick={[Function]}
        text="Remove From Project"
      />
    </div>
  </div>
  <div
    className="view-accreditations"
  >
    View accreditations
  </div>
</div>
`;

exports[`ContactTile - Should render without first and last name 1`] = `
<div
  className="row member-not-confirmed"
  id="contact-tile"
  onClick={[Function]}
>
  <div
    className="col-md-3 col-sm-12"
  >
    <div
      className="text-center avatar-container"
    >
      <img
        alt="avatar"
        className="avatar img-circle img-thumbnail"
        src="images/default-avatar.png"
      />
    </div>
  </div>
  <div
    className="col-md-9 col-sm-12"
  >
    <div
      className="row"
    >
      <h4
        className="bp3-heading"
      >
        testy.mctesterson@prind.tech
      </h4>
    </div>
    <div
      className="row"
    >
      <div
        className="col-lg-2 col-md-4"
      >
        <b>
          Email: 
        </b>
      </div>
      <div
        className="col-lg-10 col-md-8"
      >
        testy.mctesterson@prind.tech
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-lg-2 col-md-4"
      >
        <b>
          Role: 
        </b>
      </div>
      <div
        className="col-lg-10 col-md-8"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="col-lg-2 col-md-4"
      >
        <b>
          Status: 
        </b>
      </div>
      <div
        className="col-lg-10 col-md-8"
      >
        Invited
      </div>
    </div>
    <div
      className="row"
    >
      <Blueprint3.Button
        id="remove-member-button"
        intent="danger"
        onClick={[Function]}
        text="Remove From Project"
      />
    </div>
  </div>
</div>
`;

exports[`ContactTile - Should render without username 1`] = `
<div
  className="row member-not-confirmed"
  id="contact-tile"
  onClick={[Function]}
>
  <div
    className="col-md-3 col-sm-12"
  >
    <div
      className="text-center avatar-container"
    >
      <img
        alt="avatar"
        className="avatar img-circle img-thumbnail"
        src="images/default-avatar.png"
      />
    </div>
  </div>
  <div
    className="col-md-9 col-sm-12"
  >
    <div
      className="row"
    >
      <h4
        className="bp3-heading"
      >
        testy.mctesterson@prind.tech
      </h4>
    </div>
    <div
      className="row"
    >
      <div
        className="col-lg-2 col-md-4"
      >
        <b>
          Email: 
        </b>
      </div>
      <div
        className="col-lg-10 col-md-8"
      >
        testy.mctesterson@prind.tech
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-lg-2 col-md-4"
      >
        <b>
          Role: 
        </b>
      </div>
      <div
        className="col-lg-10 col-md-8"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="col-lg-2 col-md-4"
      >
        <b>
          Status: 
        </b>
      </div>
      <div
        className="col-lg-10 col-md-8"
      >
        Awaiting Sign Up
      </div>
    </div>
    <div
      className="row"
    >
      <Blueprint3.Button
        id="remove-member-button"
        intent="danger"
        onClick={[Function]}
        text="Remove From Project"
      />
    </div>
  </div>
</div>
`;
